llvm/docs/SourceLevelDebugging.rst

```diff
@@ -417,8 +448,9 @@ values through compilation, when objects are promoted to SSA values an
 ``llvm.dbg.value`` intrinsic is created for each assignment, recording the
 variable's new location. Compared with the ``llvm.dbg.declare`` intrinsic:
 
-* A dbg.value terminates the effect of any preceding dbg.values for (any
-  overlapping fragments of) the specified variable.
+* A dbg.value terminates the effect of any preceding dbg.values for any
+  spatially overlapping variables or fragments of the specified variable or
```

Comment:
The way this is worded sounds as if in this example:
```
%1 = ...
call llvm.dbg.value(metadata %1, DILocalVariable("x"), DIExpression())
%2 = ...
call llvm.dbg.value(metadata %1, DILocalVariable("y"), DIExpression())
```
The second intrinsic ends the range of "x", but that's not the case. It only terminates it if it refers to the _same_ variable or a fragment of it.

Reply:
```suggestion
  spatially overlapping llvm.dbg.values of the specified variable or
```
Perhaps this - I think given the definition of spatial overlaps, it should be clear that it applies to both fragment and non-fragment variable definitions.

Reply:
Right, it might be good to add a note here, and to make the term "spatially overlapping" be a link back to the definition, but the intention was to define "spatial overlap" such that this could just talk in terms of the universe of all variables and fragments and by definition still be restricted to the way LLVM currently works. In other words, a `dbg.value` for "x" cannot spatially overlap with a `dbg.value` for "y" by definition, so the wording here does not need to repeat that restriction.